### PR TITLE
Several updates and fixes

### PR DESCRIPTION
**Pull request by georgiosd**
- Updated npm packages and removed some redundant ones. These updates make `node-v8-clone` an optional dependency which used to break on windows.
- Fixes stuff that prevented site from running and loading
- Unit tests run again with jasmine

I haven't tested some of the other commands but if I fix any more I'll send you another PR or update this one.

### CODE REVIEW

**Comment by amcdnl** [comment]
@georgiosd  Can you merge?

**Comment by georgiosd** [comment]
It says only those with write access can merge...

I'd feel a bit more comfortable if you tried it out too - just in case - first time I touch npm stuff that has to run on non windows machines.

**Comment by amcdnl** [comment]
I mean't pull latest and rebase with changes.  @marjan-georgiev made a ton of changes this morning.

**Comment by georgiosd** [comment]
Ohhh - that makes sense, I did wonder why there were conflcts. I'll give it a shot

**Comment by marjan-georgiev** [comment]
Yes, many of our fixes overlap. Please take a look at my last two commits and apply your changes that are not there yet.

Thanks!

**Comment by georgiosd** [comment]
Done but don't merge yet - wanna update angular etc and run some more tests.

Btw - any idea how to do this with the new component router?

**Comment by georgiosd** [comment]
Should be good to go!

**Comment by amcdnl** [comment]
@marjan-georgiev can u review?

File: system.config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 System.config({
-  baseURL: "",
+  baseURL: "/",
```
Comment (marjan-georgiev) [inline system.config.js:2]: This breaks `gulp release`
Reverting it fixes it.

**Comment by marjan-georgiev** [comment]
Other than the baseURL line breaking `gulp release`, everything else looks good. @georgiosd could you take a look at that please?

**Comment by georgiosd** [comment]
Release is actually the only thing that I didn't try! Thanks for pointing that out.

Let me look and I'll get back to you on that.

**Comment by georgiosd** [comment]
Guys, it didn't seem right to change the `baseURL` to `""` for the bundling as it is a valid setting. 

I traced the error to passing `true` to `ignoreBaseURL` to `loadConfig()` on `Builder`. Not sure if this breaks anything else, I think it shouldn't.

https://github.com/Swimlane/systemjs-route-bundler/pull/39

**Comment by georgiosd** [comment]
It's not exactly sauce-tests but the E2E tests are working also

**Comment by marjan-georgiev** [comment]
I agree with that, however, for apps that have /index.html as an entry point, this is a breaking change.

`gulp release` normally only runs on the server during deployment, so any changes to it shouldn't go into source control. I did miss that commit there though, so your point is very valid :)

**Comment by georgiosd** [comment]
How about moving `index.html` to `src` and dealing with it that way? All input in `src`, all output in `dist`

**Comment by amcdnl** [comment]
I'm not really a fan; this makes like 3 entry points: src, dist, release.  I tried this at first, but it just felt really odd and had a few problems though i dont remember now.

You should never be checking in the index.html release mode version.  That is only ran on the build server like @marjan-georgiev mentioned.

**Comment by georgiosd** [comment]
What do you mean by 3 entry points? `src` is the original file, `dist` is after transformation (copy) and `release` is after transformation (bundle).

Are you saying that because `serve` would have to pick between `dist/index.html` or `release/index.html` ? 

We could have two tasks: `serve` and `serve-release` which requires `release`. At the end of the day `serve` is used in conjunction with `watch` for dev - `serve-release` would only be to make sure `release` is not broken.

**Comment by m0ppers** [comment]
Just stumbled about the index.html stuff.

IMHO it is totally evil to modify the src index.html (and an easy cause for trouble as it is very likely to be commited in the prod state - just happened ;) ).

So I agree with @georgiosd (if I understood him correctly). The index.html should be copied and then modified inside the dist folder and the src index should be untouched.

It should be possible to simply scp the dist folder to some web server serving static files. Also it should be totally possible that the webserver serves the files in a subdirectory. So any hardcoded baseURL would be evil as well IMHO. So including should always be done relatively.

I wouldn't necessarily move index.html to src as this would make it more difficult to jump into creating an application with this seed.

My 2 cents :O

**Comment by marjan-georgiev** [comment]
@georgiosd could we please create a separate pull request for that change and move our discussion there? I'd like to get your other fixes merged in as soon as possible.

**Comment by georgiosd** [comment]
@marjan-georgiev how would I do that? Remove that commit from history and force push?

**Comment by marjan-georgiev** [comment]
Just push another commit with that change reverted. 

Then after I merge this pull request, branch off of master and create another pull request with it.

**Comment by georgiosd** [comment]
Ok

**Comment by zeeshanjan82** [comment]
I have a related comment I am not sure if it helps but I need some help in this. I have upgraded my NodeJS version to 4.1.1 and when I  run gulp watch I have issue with node-v8-clone package as it is dependent on the older versions on NodeJS. Any idea how can I run the seed with NodeJS version 4.1.1

**Comment by marjan-georgiev** [comment]
@zeeshanjan82 node-v8-clone is no longer a dependency. After this pull request is merged I'll update the project to use the latest systemjs-route-bundler, and that should fix it.

**Comment by zeeshanjan82** [comment]
Thanks @marjan-georgiev I will await the response on this merge.

**Comment by georgiosd** [comment]
@marjan-georgiev you're good to go

**Comment by georgiosd** [comment]
I'll create another pull request for the dist change later on

**Comment by marjan-georgiev** [comment]
@georgiosd perfect, thank you!

**Comment by georgiosd** [comment]
New pull request and discussion here: https://github.com/Swimlane/angular-systemjs-seed/pull/50 @m0ppers @zeeshanjan82

**Pull request merged**